static/app/components/arithmeticBuilder/validator.tsx

```diff
@@ -19,106 +21,172 @@ export function validateTokens(tokens: Token[]): boolean {
   return validator.isValid();
 }
 
-enum Part {
-  OPEN_PAREN = '(',
-  OPERATOR = 'o',
-  EXPRESSION = 'e',
+export function computeNextAllowedTokenKinds(tokens: Array<Token | null>): TokenKind[][] {
+  const validator = new ExpressionValidator();
+  const tokenKinds = [];
+
+  for (const token of tokens) {
+    if (!defined(token)) {
+      tokenKinds.push([]);
+    } else if (isTokenFreeText(token)) {
+      tokenKinds.push(validator.nextAllowedTokenKinds());
```

Comment:
ok so basically since after every token, there's a free text token, we essentially only push token suggestions for those?

can we just add some comments here so it's easy to get context if we have to come back to the code